### DISCUSSION TITLE
Saving ini

### DISCUSSION THREAD

**Discussion by Boomslangnz** [ocornut/imgui#4571] [New Users Build/Link/Run issues ONLY!]
Hi, i do apologize if i am in wrong section.

Is there a way to save the window positions etc to ini file without wiping anything else in ini?

I have bunch of values getting saved to my ini like colors and my checkboxes but if i make imgui save the window positons etc on same ini it will wipe all my other values each time i move window (imgui saves to ini)

Thanks

**Comment by ocornut** [comment]
You can set `io.IniFilename` to NULL to disable writing to file, then use `LoadIniSettingsFromMemory()` and `SaveIniSettingsToMemory()`  to save into a buffer and then write the ini file yourself.

Or you can register a `ImGuiSettingsHandler` and read/write the contents of your ini data while dear imgui read/write its own.
